src/Resolvers/Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver/WorkloadSdkResolver.cs

```diff
@@ -1,145 +1,139 @@
-                List<string> workloadManifestPaths = new List<string>();
-                foreach (var manifestDirectory in _workloadManifestProvider.GetManifestDirectories())
+#if USE_SERILOG
+                logActions.Add(log => log.Information("Initializing resolver state"));
+#endif
+                var dotnetRootPath = GetDotNetRoot(resolverContext);
+
+                var sdkDirectory = GetSdkDirectory(resolverContext);
+                //  The SDK version is the name of the SDK directory (ie dotnet\sdk\5.0.100)
+                var sdkVersion = Path.GetFileName(sdkDirectory);
+
+                cachedState = new CachedState()
                 {
-                    var workloadManifestTargetPath = Path.Combine(manifestDirectory, "WorkloadManifest.targets");
-                    if (File.Exists(workloadManifestTargetPath))
-                    {
-                        workloadManifestPaths.Add(manifestDirectory);
-                    }
-                }
-                return factory.IndicateSuccess(workloadManifestPaths, sdkReference.Version);
+                    DotnetRootPath = dotnetRootPath,
+                    SdkVersion = sdkVersion,
+                    WorkloadResolver = new WorkloadPartialResolver()
+                };
+
+                resolverContext.State = cachedState;
             }
             else
             {
-                var packInfo = _workloadResolver.TryGetPackInfo(sdkReference.Name);
-                if (packInfo != null)
-                {
-                    if (Directory.Exists(packInfo.Path))
-                    {
-                        return factory.IndicateSuccess(Path.Combine(packInfo.Path, "Sdk"), sdkReference.Version);
-                    }
-                    else
-                    {
-                        var itemsToAdd = new Dictionary<string, SdkResultItem>();
-                        itemsToAdd.Add("MissingWorkloadPack",
-                            new SdkResultItem(sdkReference.Name,
-                                metadata: new Dictionary<string, string>()
-                                {
-                                    { "Version", packInfo.Version }
-                                }));
-
-                        Dictionary<string, string> propertiesToAdd = new Dictionary<string, string>();
-                        return factory.IndicateSuccess(Enumerable.Empty<string>(),
-                            sdkReference.Version,
-                            propertiesToAdd: propertiesToAdd,
-                            itemsToAdd: itemsToAdd);
-                    }
-                }
+#if USE_SERILOG
+                logActions.Add(log => log.Information("Using cached resolver state"));
+#endif
             }
-            return null;
+
+            var result = cachedState.WorkloadResolver.Resolve(sdkReference.Name, cachedState.DotnetRootPath, cachedState.SdkVersion
+#if USE_SERILOG
+                , logActions
+#endif
+                );
+
+
+#if USE_SERILOG
```

Comment:
Any way to make this "more repeatable"? If another dev wants to improve the performance again. How should the new dev know what these means and how these properties should be used? What is the existing base line?

It is hard to make a unit test. But could you have a write up on how to use these logs? Something like you need to enable EnableSerilogLogging, copy the files over, and when running in VS then we should expect MSBuildSubmissionId to be the same all the time?